app/javascript/components/editor/TestRunOutput.tsx

```diff
@@ -1,18 +1,33 @@
 import React from 'react'
 import { TestRun } from './types'
-import { TestsGroupList } from './TestsGroupList'
+import { TestsGroupedByStatusList } from './TestsGroupedByStatusList'
+import { TestsGroupedByTaskList } from './TestsGroupedByTaskList'
 
 export const TestRunOutput = ({
   testRun,
 }: {
   testRun: TestRun
 }): JSX.Element => {
-  return testRun.version === 2 || testRun.version === 3 ? (
-    <TestsGroupList
-      tests={testRun.tests}
-      language={testRun.highlightjsLanguage}
-    />
-  ) : (
+  if (testRun.version >= 3 && testRun.tasks.length > 0) {
+    return (
+      <TestsGroupedByTaskList
+        tests={testRun.tests}
+        language={testRun.highlightjsLanguage}
+        tasks={testRun.tasks}
+      />
+    )
+  }
+
+  if (testRun.version >= 2) {
```

Comment:
I consider adding a `&& testRun.tests.length > 0` check here too. Might be a good idea, as otherwise you'd get an empty list that might look weird

Reply:
Add it.